### PR TITLE
[Data][Docs] Add "Working with Images" guide

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

Image is a common data format for Data users. This PR adds a guide tailored specifically for this use case.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/working-with-images.rst [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+--------------
+
+Ray Data can read images from a variety of formats.
+
+To view the full list of supported file formats, see the
+:ref:`Input/Output reference <input-output>`.
+
+.. tab-set::
+
+    .. tab-item:: Raw images
```
Comment (amogkam) [inline doc/source/data/working-with-images.rst:27]: I would clarify this as "Ray images (JPG, PNG, etc.)" for full clarification

File: doc/source/data/working-with-images.rst [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+        To load raw images, call :func:`~ray.data.read_images`.
```
Comment (amogkam) [inline doc/source/data/working-with-images.rst:29]: Let's explicitly say that the PIL library is used and any format supported by PIL would work?

File: doc/source/data/working-with-images.rst [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+For more information on performing inference, see
+:ref:`End-to-end: Offline Batch Inference <batch_inference_home>`
+and :ref:`Transforming batches with actors <transforming_data_actors>`.
+
+.. _saving_images:
+
+Saving images
+-------------
+
+Save images in Parquet or Numpy files. Other formats aren't supported.
```
Comment (amogkam) [inline doc/source/data/working-with-images.rst:240]: Do we need to say other formats aren't supported?

Comment (amogkam) [reply doc/source/data/working-with-images.rst:240]: Let's just link to the input/output ref instead?

File: doc/source/data/user-guide.rst [hunk 4]
```diff
@@ -18,6 +18,7 @@ show you how achieve several tasks.
     iterating-over-data
     saving-data
     working-with-tensors
+    working-with-images
```
Comment (amogkam) [inline doc/source/data/user-guide.rst:21]: I would move this above working with text

File: doc/source/data/working-with-images.rst (hunk 1 above)
Comment (bveeramani) [reply doc/source/data/working-with-images.rst:27]: Tried this but felt that the tab text was too verbose. Clarified in the content instead.

Lemme know what you think.

File: doc/source/data/working-with-images.rst (hunk 3 above)
Comment (bveeramani) [reply doc/source/data/working-with-images.rst:240]: What would users find in the API references other than `write_parquet` and `write_numpy`?

File: doc/source/data/working-with-images.rst (hunk 1 above)
Comment (bveeramani) [reply doc/source/data/working-with-images.rst:27]: We could also rename "Raw images" as "Image files"

File: doc/source/data/working-with-images.rst [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+.. tab-set::
+
+    .. tab-item:: Raw images
+
+        To load raw images like JPEG files, call :func:`~ray.data.read_images`.
+
+        .. note::
+
+            :func:`~ray.data.read_images` uses
+            `Pillow <https://pillow.readthedocs.io/en/stable/index.html>`_. For a list of
```
Comment (amogkam) [inline doc/source/data/working-with-images.rst:34]: I would call this PIL. I think this is what most people refer to it as, for example in the pytorch docs.

Comment (amogkam) [inline doc/source/data/working-with-images.rst:34]:
```suggestion
            `PIL <https://pillow.readthedocs.io/en/stable/index.html>`_. For a list of
```

**Review by amogkam — APPROVED** [review]
1 small comment

File: doc/source/data/working-with-images.rst (hunk 3 above)
Comment (bveeramani) [reply doc/source/data/working-with-images.rst:247]: Discussed offline -- `write_json` also works. Will update accordingly

**Pull request merged**